### DISCUSSION TITLE
SPARQL endpoint for querying multiple datasets

### DISCUSSION THREAD

**Discussion by royassis** [apache/jena#2287] [Q&A]
Hey there

I want to build a cross dataset SPRAQL query. 
I am trying to use the `FROM` clause but it seems that the **SPARQL endpoint** is set to only one dataset.

![image](https://github.com/apache/jena/assets/34189329/78e79c86-1042-484f-afba-99c78fa13e6c)

Does Fuseki server have a endpoint where I can submit cross dataset queries ?

**Comment by flange-ipb** [comment]
Hi @royassis,

the `FROM` keyword may be used for selection of a named graph inside the dataset you are querying.
A _cross dataset query_ should be doable via a [SPARQL federated query](https://www.w3.org/TR/sparql11-federated-query/). In particular, querying a dataset on the same Fuseki instance can be achieved by using `http://127.0.0.1:3030/datasetname/` as the target endpoint.

Example:
```
SELECT *
WHERE {
  SERVICE <http://127.0.0.1:3030/datasetname> {
    ?s ?p ?o
  }
}
LIMIT 10
```

**Reply by royassis** [reply]
Hey @flange-ipb thanks for the response. I used your suggestion. 

Regarding the `FROM` and `FROM NAMED` keywords.
In the [RDF spec](https://www.w3.org/TR/sparql11-query/#specifyingDataset) they mention FROM and FROM named in this context:

> A SPARQL query may specify the dataset to be used for matching by using the FROM clause and the FROM NAMED clause to describe the RDF dataset

I'm still a bit confused.

**Reply by afs** [reply]
"describe the dataset" here means enumerate the graphs to have in the dataset.

FROM and FROM identify the graphs to go into a dataset that is then queried.
